### DISCUSSION TITLE
I need help to make sure

### DISCUSSION THREAD

**Discussion by ahmed-syntax** [ultralytics/ultralytics#16432] [Q&A]
```
from ultralytics import YOLO

# load a pretrained custom model

model = YOLO("custom.pt")  

# new data set to feed the model

results = model.train(data="new_custom_data.yaml", epochs=100, imgsz=640) 
```

is this will continue the model to train with the new data ? without harming the old training data that i trained my model before?

Second question if i have only training data set how i can add valid data to the .yaml file

**Comment by UltralyticsAssistant** [comment]
👋 Hello @balg0rithm, thank you for reaching out to Ultralytics 🚀! Your question is interesting, and while this is an automated response, an Ultralytics engineer will assist you soon.

We recommend checking out the [Docs](https://docs.ultralytics.com) where you can find [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) examples. These might help clarify your questions and ensure you're on the right track.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/). That will greatly help in diagnosing any issues you encounter.

For questions related to custom training ❓, please include detailed information, like dataset images and logs. Make sure you're following our [Model Training Tips](https://docs.ultralytics.com/guides/model-training-tips/).

Join our community in various spaces:
- For real-time interactions, check out our [Discord](https://ultralytics.com/discord) 🎧.
- Participate in in-depth discussions on [Discourse](https://community.ultralytics.com).
- Engage with peers on our [Subreddit](https://reddit.com/r/ultralytics).

## Upgrade

Ensure your `ultralytics` package is up to date. Use a [**Python>=3.8**](https://www.python.org/) environment and upgrade with:

```bash
pip install -U ultralytics
```

## Environments

You can run YOLOv8 in any of these environments with preinstalled dependencies:

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/ultralytics/yolov8"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud**: [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon**: [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker**: [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml/badge.svg" alt="Ultralytics CI"></a>

A green badge indicates all tests are passing, ensuring YOLOv8 operates correctly across platforms.

**Comment by glenn-jocher** [comment]
@balg0rithm yes, using `model.train(data="new_custom_data.yaml", epochs=100, imgsz=640)` will continue training with the new data without affecting the original model. For validation data, specify the path in your `.yaml` file under the `val` key.

**Reply by ahmed-syntax** [reply]
But I'm loading a pre-trained model with that 

```
Model=YOLO("pretrained.pt")
```
And I'm getting bad results the new data affect the old ones and sometimes override the old data

**Reply by ahmed-syntax** [reply]
someone told me catastrophic forgetting ?

Should i combine? The two dataset!
I want to make my model Smarter but I don't have the data available i train it on data and after that i have new data I want to train it too can u tell me what to do in this case?

**Reply by glenn-jocher** [reply]
To mitigate catastrophic forgetting, consider using techniques like fine-tuning with a balanced dataset that includes samples from both the old and new data. If you don't have access to the old data, you might explore methods like knowledge distillation or incremental learning.
